tests/build/test_utils.py

```diff
@@ -108,3 +111,110 @@ def test_process_accelerate_launch_custom_config_file(patch_path_exists):
     temp_job_config = {"accelerate_launch_args": {"config_file": dummy_config_path}}
     args = process_accelerate_launch_args(temp_job_config)
     assert args.config_file == dummy_config_path
+
+
+class CopyCheckpointTestConfig:
+
+    def __init__(self, test_root):
+
+        assert os.path.isdir(test_root)
+
+        self.test_root = test_root
+        self.source_dir = os.path.join(self.test_root, "test_copytree_source")
+        self.source_sub_dir = os.path.join(self.source_dir, "subdir1")
+
+        if os.path.isdir(self.source_dir):
+            shutil.rmtree(self.source_dir)
+
+        os.mkdir(self.source_dir)
+        open(os.path.join(self.source_dir, "tf1.txt"), "a").close()
+        open(os.path.join(self.source_dir, "tf2.txt"), "a").close()
+        open(os.path.join(self.source_dir, "tf3.txt"), "a").close()
+
+        os.mkdir(self.source_sub_dir)
+        open(os.path.join(self.source_sub_dir, "tf4.txt"), "a").close()
+        open(os.path.join(self.source_sub_dir, "tf5.txt"), "a").close()
+        open(os.path.join(self.source_sub_dir, "tf6.txt"), "a").close()
+
+    def are_dir_trees_equal(self, dir1, dir2):
+
+        dirs_cmp = filecmp.dircmp(dir1, dir2)
+        if (
+            len(dirs_cmp.left_only) > 0
+            or len(dirs_cmp.right_only) > 0
+            or len(dirs_cmp.funny_files) > 0
+        ):
+            return False
+        (_, mismatch, errors) = filecmp.cmpfiles(
+            dir1, dir2, dirs_cmp.common_files, shallow=False
+        )
+        if len(mismatch) > 0 or len(errors) > 0:
+            return False
+        for common_dir in dirs_cmp.common_dirs:
+            new_dir1 = os.path.join(dir1, common_dir)
+            new_dir2 = os.path.join(dir2, common_dir)
+            if not self.are_dir_trees_equal(new_dir1, new_dir2):
+                return False
+        return True
+
+
+def test_copy_checkpoint_dest_dir_does_not_exist():
+
+    # Init source directory
+    config = CopyCheckpointTestConfig("/tmp")
+
+    target_dir_does_not_exist = os.path.join(config.test_root, "test_copytree_target")
+
+    # Execute the copy
+    copy_checkpoint(config.source_dir, target_dir_does_not_exist)
+    assert config.are_dir_trees_equal(config.source_dir, target_dir_does_not_exist)
```

Comment:
I was hopeful that this could be done with something like...
```py
assert list(pathlib.Path(source).rglob("*")) ==  list(pathlib.Path(dest).rglob("*"))
```
but alas, my code fails because the root directories are different and it doesn't check the contents of the files either. Unfortunate :/

Reply:
The only alternative checking I was thinking of is...
```py
for root,dir,files in os.walk(source_dir):
     new_root = root.replace(source_dir, dest_dir)
     for file in files:
             assert os.path.exists(os.path.join(new_root, file))
```
but is not perfect either, so I think if yours works, good to go with it!